### ISSUE TITLE
Notification personalization concept (colors, combination of LEDs flashing)

### ISSUE THREAD

**Issue by cpfeiffer** [Freeyourgadget/Gadgetbridge#37]
This is related to #29: we should have a personalization concept for the notifications. The user should be able to list a few favourite contacts (or even groups?) from the addressbook and configure a special notification for those.

For a start, the user should be able to specify a colour for a contact/contact group. All notifications coming from that contact/group shall then use that colour.

**Comment by ov3rk1ll** [comment]
I was looking a bit into changing the LED color and I'm testing with
```java
new byte[] { 14, r, g, b, 1 };
```

I've found out two things while playing around with this:
 - r, g, b and must be between 0-6 which limits the colors a bit but I think there are still enough options  - see https://dl.dropboxusercontent.com/u/35486665/Android/mi/colormap.html
 - this only affects the color for the "progress to daily goal" LEDs, notification still show up with 3 greens (byte[] { 8, 0 };) or center-only blue (byte[] { 8, 1 };).

Has anyone find a way to turn on LEDs individually or even with different colors?

**Comment by danielegobbetti** [comment]
I know that this app: https://github.com/UgoRaffaele/miband-notifier allows the user to set a specific color for different notifications, additionally it blinks all three leds when receiving a notification, hence perhaps it's using the "goal reached" notification?

**Comment by ov3rk1ll** [comment]
@danielegobbetti yes, it seems be doing that (https://github.com/UgoRaffaele/miband-notifier/blob/master/src/main/java/com/ugopiemontese/openband/MiBandCommunicationService.java#L241) so it appears that there is no way to change the "default" notification color/style

**Comment by cpfeiffer** [comment]
So you can either have custom vibration **or** custom color :-/

**Comment by arnottcr** [comment]
I am quite interested in this feature, as the codebase that you reference, and [its parent](https://github.com/lwis/miband-notifier), are both getting a little dated and do not support the new devices (like the 1S and new 1 firmwares).

At the moment lwis' application run side-by-side gadget bridge with no issue, so the integration should not be too painful (maybe).

**Comment by cpfeiffer** [comment]
I got a little demotivated with this feature, because the new models have white LEDs only, so you can't use fancy colours anymore. Also the protocol changed, making it harder or even impossible to use custom vibration patterns.

Off the top of my head I can't say what kind of custom patterns are still possible. AFAIR, the standard protocol only specifies something like three different alert levels which we can choose.

Would need some time to investigate the possibilities with current devices and firmware versions.

**Comment by arnottcr** [comment]
I recently got a friend a miband 1S and was sad to confirm that the LEDs are all white. It seems like a strange feature regression.

My concern and reasons for the bump on this request are mostly because that code-base has become quite stagnant. My solution is workable for the near foreseeable future, so no worries, but I thought I would see what the current state is and it looks bleak.

It looks like some type of look up table to convert firmware version to feature set may be required. (shudder)

**Comment by geripgeri** [comment]
Have there been any decisions regarding the launch of the custom led colorization feature? Is it under development?

**Comment by cpfeiffer** [comment]
No, nothing has changed since then (see my previous comment from April 14th).

If anyone wants to work on this, I'm with patches, but I personally won't spend any time on this in the near future. This might change when we get to support another device that has more options for custom notifications.
